File: Dread's Projects/DreadRPG/GameStates/MainMenuState.cpp
#include "MainMenuState.h"

MainMenuState::MainMenuState(GameData* gData): GameState(gData) {}

void MainMenuState::ShowCredits()
{
    Interface::PrintCredits();
}

void MainMenuState::ShowHelpSection()
{
    Interface::PrintHelpSection();
}

void MainMenuState::ShowMenu()
{
    Interface::PrintMainMenu();
}

MainMenuState::~MainMenuState() {}

File: Java/Monster Slayer Java Edition/src/GameLogic/GameStates/MainMenuState.java
package GameLogic.GameStates;

import GameLogic.GameData;

public class MainMenuState extends StateAdapter
{
    private GameData gData;
    
    public MainMenuState()
    {
        super();
        this.gData = new GameData();
    }
    
    public MainMenuState(GameData gData)
    {
        super();
        this.gData = gData;
    }

    @Override
    public GameState handleInput(char input)
    {
        switch(input)
        {
            case '1':
                return new GetPlayerNameState(gData);
            case '2':
                return new CreditsState();
            case '3':
                return new ExitState();
            default:
                return this;
        }
    }
    
    @Override
    public String getStateName() { return "MainMenuState"; }
}